sootup.java.core/src/main/java/sootup/java/core/interceptors/typeresolving/TypePromotionVisitor.java

```diff
@@ -56,6 +56,13 @@ public static boolean isIntermediateType(Type type) {
   }
 
   public void visit(@Nonnull Value value, @Nonnull Type stdType, @Nonnull Stmt stmt) {
+
+    /* Note: When visiting function parameters, we may encounter constant values such as strings ("abc") or integers (2).
+      These constants are not instances of the Local class and should be handled accordingly.
+    */
+    if (!(value instanceof Local)) {
+      return;
+    }
```

Comment:
Can you please make the comment more specific that we are possibly over-approximating here and if things break in future it might be due to this.